src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -389,6 +444,71 @@ private static string GetEscapedParameterName(AttributeData attribute, string pa
         }
     }
 
+    private static bool TryGetAsParametersConstructor(INamedTypeSymbol type, out bool? isParameterlessConstructor, [NotNullWhen(true)] out IEnumerable<ConstructorParameter>? matchedParameters)
+    {
+        isParameterlessConstructor = null;
+        matchedParameters = null;
+        if (type.IsAbstract)
+        {
+            return false;
+        }
+
+        var constructors = type.Constructors.Where(constructor => constructor.DeclaredAccessibility == Accessibility.Public && !constructor.IsStatic);
+        var numOfConstructors = constructors.Count();
+
+        if (numOfConstructors == 1)
+        {
+            var targetConstructor = constructors.SingleOrDefault();
+            var properties = type.GetMembers().OfType<IPropertySymbol>().Where(property => property.DeclaredAccessibility == Accessibility.Public);
+            var lookupTable = new Dictionary<ParameterLookupKey, IPropertySymbol>();
+            foreach (var property in properties)
+            {
+                lookupTable.Add(new ParameterLookupKey(property.Name, property.Type), property);
+            }
+
+            var parameters = targetConstructor.GetParameters();
+            var propertiesWithParameterInfo = new List<ConstructorParameter>();
+
+            if (parameters.Length == 0)
+            {
+                isParameterlessConstructor = true;
+                matchedParameters = properties.Select(property => new ConstructorParameter(property, null));
+                return true;
+            }
+
+            foreach (var parameter in parameters)
+            {
+                var key = new ParameterLookupKey(parameter.Name!, parameter.Type);
+                if (lookupTable.TryGetValue(key, out var property))
+                {
+                    propertiesWithParameterInfo.Add(new ConstructorParameter(property, parameter));
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            isParameterlessConstructor = false;
+            matchedParameters = propertiesWithParameterInfo;
+            return true;
+        }
+
+        if (type.IsValueType)
+        {
+            isParameterlessConstructor = true;
+            matchedParameters = type.GetMembers().OfType<IPropertySymbol>().Select(property => new ConstructorParameter(property, null));
```

Comment:
Do we need to filter out "get-only" properties here? Basically if the property doesn't have a setter, then what?

Also, above in the `numOfConstructors == 1` block, we are looking at accessibility.

Reply:
Good catch indeed. I've made the necessary tweaks.